R/bayesianpredictionperformance.R

```diff
@@ -0,0 +1,269 @@
+#
+# Copyright (C) 2019 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+BayesianPredictionPerformance  <- function(jaspResults, dataset, options, state = NULL) {
+
+  options[["method"]] <- "BAYES"
+  ready <- .metamiscReady(options)
+
+  if (ready) {
+    dataset <- .metamiscGetData(options, dataset)
+    .metamiscFitModelBayesian(jaspResults, options, dataset)
+  }
+
+  .metamiscSummaryTable(jaspResults, options)
+
+  if (jaspResults[["summaryTable"]]$getError())
+    return()
```

Comment:
This is not what I meant about the error handling, it is meaningless to do return here.

Reply:
I do not really see the difference between these two solutions apart from the fact that I will need to write ```jaspResults[["summaryTable"]]$getError()``` 4 more times in classical and 8 more times in Bayesian version. If the model can't be fit, the remaining options should become unavailable - both cases make that happen.

Reply:
Here's an example. Without this, no matter what options you select, once there is an error this is what the user gets:

![image](https://user-images.githubusercontent.com/21319932/135044167-209c3d7a-c2bb-4c68-bf90-d8b6cfe78080.png)

In other words, once there is an error it looks like the analysis is completely unresponsive. That is a bad user experience.

If you do as I suggest, a user will still see empty output elements are still whenever they select the corresponding options:

![image](https://user-images.githubusercontent.com/21319932/135045187-18e4b620-6179-4802-a2b4-b5bff1ac9353.png)

Had you put the error on jaspResults (or any common container), then the empty plot element would also be shown greyed out like the table, instead of black.

See also https://github.com/jasp-stats/jasp-desktop/blob/development/Docs/development/r-analyses-guide.md#propagating-errors-through-a-container

Reply:
ok, I see the point now. I didn't expect that we want to show empty figures when the whole analysis is crashed. I changed it appropriately now.